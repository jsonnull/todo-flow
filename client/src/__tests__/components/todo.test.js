import React from 'react';
import { render } from 'react-dom';
import renderer from 'react-test-renderer';

import Todo from '../../components/todo';

describe('Todo', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    render(<Todo />, div);
  });

  it('puts a line through text when completed', () => {
    const component = renderer
      .create(<Todo text="Test Content" completed />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('calls onClick when clicked', () => {
    const onClick = jest.fn();
    const component = renderer.create(<Todo onClick={onClick} />).toJSON();
    component.props.onClick();

    expect(component).toMatchSnapshot();
    expect(onClick).toHaveBeenCalled();
  });
});
