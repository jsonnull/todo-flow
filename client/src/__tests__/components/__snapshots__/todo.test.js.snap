exports[`Todo calls onClick when clicked 1`] = `
<li
  onClick={[Function]}
  style={
    Object {
      "textDecoration": "none",
    }
  } />
`;

exports[`Todo puts a line through text when completed 1`] = `
<li
  onClick={undefined}
  style={
    Object {
      "textDecoration": "line-through",
    }
  }>
  Test Content
</li>
`;
